src/App.tsx

```diff
@@ -3,33 +3,61 @@ import './App.scss';
 import { MoviesList } from './components/MoviesList';
 import moviesFromServer from './api/movies.json';
 
-type State = {};
+interface Movie {
+  title: string,
+  description: string,
+  imgUrl: string,
+  imdbUrl: string,
+  imdbId: string,
+}
+
+type State = {
+  moviesList: Movie[]
+  searchParam: string,
+};
 
 export class App extends React.Component<{}, State> {
-  state: State = {};
+  state: State = {
+    moviesList: moviesFromServer,
+    searchParam: '',
+  };
 
   render() {
+    const { moviesList, searchParam } = this.state;
+
+    let visibleMovies = moviesList;
+
+    if (searchParam.length > 0) {
```

Comment:
you really need this check?
You can use like:
const visibleMovies = moviesList.filter...
If length of searchParam is 0 then you will see all of possible movies.

Reply:
agreed